### ISSUE TITLE
current() in XPath predicate may not be working correctly

### ISSUE THREAD

**Issue by MartijnR** [enketo/enketo-core#343]
https://github.com/kobotoolbox/enketo-express/issues/347

**Comment by MartijnR** [comment]
Looks like a more serious issue. E.g. in Chris' form this returns empty result when value is present:

`form.ex('../survey_region', 'string', '/samplecascading/testgroup/survey_country', 0, true)`

**Comment by MartijnR** [comment]
requires extra step down:

`form.ex('../../survey_region', 'string', '/samplecascading/testgroup/survey_country', 0, true)`

**Comment by MartijnR** [comment]
similarly: 

`form.ex('./survey_country2', 'nodes', '/samplecascading/testrepeat/testgroup2/survey_country3', 0, true)`

to 

`form.ex('../survey_country2', 'nodes', '/samplecascading/testrepeat/testgroup2/survey_country3', 0, true)`

**Comment by MartijnR** [comment]
when current() is used inside a predicate it goes back one step by itself. Very weird. When used outside a predicate, e.g. in a constraint, it works normally.

**Comment by MartijnR** [comment]
I think the smart way to fix #2 is to replace current() with /model/instance[1]/path/to/current/node, so e.g. in a /path/to/node

```
current()/../../anothernode
```
would become something like:

```
/model/instance[1]/path/to/node/../../anothernode
``

Also need to inject repeat positions if that is not handled by FormModel.prototype.makeBugCompliant already.

**Comment by MartijnR** [comment]
#1 is indeed a javarosa bug: https://groups.google.com/forum/#!topic/opendatakit-developers/nN8HTA5pUIM
